src/Eviden.VirtualGrocer/Client/ConfigurationLoader.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Eviden.VirtualGrocer.Shared.Models;
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+using Newtonsoft.Json;
+
+namespace Eviden.VirtualGrocer.Web.Client
+{
+    /// <summary>
+    /// Extension method container class for loading configuration from the server.
+    /// </summary>
+    public static class ConfigurationLoader
+    {
+        /// <summary>
+        /// Loads the client configuration from the server and sets up authentication.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns></returns>
+        /// <exception cref="ApplicationException"></exception>
+        public static async Task<WebAssemblyHostBuilder> LoadConfigurationAndSetupAuthenticationAsync(this WebAssemblyHostBuilder builder)
+        {
+            var configuration = builder.Configuration.Build();
+            var httpClient = new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) };
+
+            var response = await httpClient.GetAsync("/clientSettings");
```

Comment:
You can call and deserialize in one call with httpClient.GetFromJsonAsync<ClientSettings>("/clientSettings");
```suggestion
            var settings = await httpClient.GetFromJsonAsync<ClientSettings>("/clientSettings");
```